src/client.ts

```diff
@@ -246,27 +246,32 @@ export class AwaitConnection {
   }
 
   private _parseDescribeTypeMessage(): [number, ICodec, ICodec] {
-    this._rejectHeaders();
+    try {
+      this._rejectHeaders();
 
-    const cardinality: char = this.buffer.readChar();
+      const cardinality: char = this.buffer.readChar();
 
-    const inTypeId = this.buffer.readUUID();
-    const inTypeData = this.buffer.readLenPrefixedBuffer();
-    let inCodec = this.codecsRegistry.getCodec(inTypeId);
-    if (inCodec == null) {
-      inCodec = this.codecsRegistry.buildCodec(inTypeData);
-    }
+      const inTypeId = this.buffer.readUUID();
+      const inTypeData = this.buffer.readLenPrefixedBuffer();
+      let inCodec = this.codecsRegistry.getCodec(inTypeId);
+      if (inCodec == null) {
+        inCodec = this.codecsRegistry.buildCodec(inTypeData);
+      }
 
-    const outTypeId = this.buffer.readUUID();
-    const outTypeData = this.buffer.readLenPrefixedBuffer();
-    let outCodec = this.codecsRegistry.getCodec(outTypeId);
-    if (outCodec == null) {
-      outCodec = this.codecsRegistry.buildCodec(outTypeData);
-    }
+      const outTypeId = this.buffer.readUUID();
+      const outTypeData = this.buffer.readLenPrefixedBuffer();
+      let outCodec = this.codecsRegistry.getCodec(outTypeId);
```

Comment:
I'd do it the other way:

* parsing of the protocol message and `this.buffer.finishMessage();` should be outside the `try` block; 
* `this.codecsRegistry.getCodec(outTypeId);` should be after the parsing -- and then you probably don't need `try..catch` at all